src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -379,6 +384,48 @@ void FFN<OutputLayerType, InitializationRuleType>::Serialize(
   }
 }
 
+template<typename OutputLayerType, typename InitializationRuleType>
+void FFN<OutputLayerType, InitializationRuleType>::TrivialCopy(const NetworkType& network)
+{
+  // Copy variables from source network
+  outputLayer = network.outputLayer;
+  initializeRule = network.initializeRule;
+  width = network.width;
+  height = network.height;
+  reset = network.reset;
+  predictors = network.predictors;
+  responses = network.responses;
+  parameter = network.parameter;
+  numFunctions = network.numFunctions;
+  error = network.error;
+  currentInput = network.currentInput;
+  currentTarget = network.currentTarget;
+}
+
+template<typename OutputLayerType, typename InitializationRuleType>
+FFN<OutputLayerType, InitializationRuleType>::FFN(const FFN& network)
+{
+  // Build new layers according to source network
+  TrivialCopy(network);
+  for (size_t i = 0; i < network.network.size(); ++i)
+  {
+    this->network.push_back(boost::apply_visitor(copyVisitor, network.network[i]));
+  }
+};
+
+template<typename OutputLayerType, typename InitializationRuleType>
+FFN<OutputLayerType, InitializationRuleType>&
+FFN<OutputLayerType, InitializationRuleType>::operator = (const FFN& network)
+{
+  // Replicate layers from source network
```

Comment:
We should deconstruct the network first, to avoid memory leaks. Also maybe it's a good idea to use the  copy-and-swap idiom here, could also be useful if we provide a move constructor, what do you think?